apps/rush-lib/src/cli/taskRunner/ProjectTask.ts

```diff
@@ -163,29 +158,30 @@ export class ProjectTask implements ITaskDefinition {
           this._hasWarningOrError = true;
         });
 
-        task.on('close', (code: number) => {
-          // Write the logs to disk
-          this._writeLogsToDisk(writer);
-
-          if (code) {
-            reject([new TaskError('error', `Returned error code: ${code}`)]);
-          } else if (this._hasWarningOrError) {
-            resolve(TaskStatus.SuccessWithWarning);
-          } else {
-            // Write deps on success.
-            if (currentPackageDeps) {
-              JsonFile.save(currentPackageDeps, currentDepsPath);
+        return new Promise((resolve: (status: TaskStatus) => void, reject: (errors: TaskError[]) => void) => {
+          task.on('close', (code: number) => {
+            // Write the logs to disk
```

Comment:
What if the stream gets closed before the event handler is attached?  Is there a way we could avoid that?

Reply:
Looks like that works fine.

---
In reply to: [185681920](https://github.com/Microsoft/web-build-tools/pull/636#discussion_r185681920) [](ancestors = 185681920)